UIInfoSuite2/UIElements/ShowQueenOfSauceIcon.cs

```diff
@@ -164,92 +95,103 @@ private void OnRenderingHud(object sender, RenderingHudEventArgs e)
+                //        new Vector2(iconLocation.X, iconLocation.Y),
+                //        new Rectangle(306, 291, 14, 14),
+                //        Color.White,
+                //        0,
+                //        Vector2.Zero,
+                //        scale,
+                //        SpriteEffects.None,
+                //        1f);
+
+                //    ClickableTextureComponent texture =
+                //        new ClickableTextureComponent(
+                //            _gus.Name,
+                //            new Rectangle(
+                //                iconLocation.X - 7,
+                //                iconLocation.Y - 2,
+                //                (int)(16.0 * scale),
+                //                (int)(16.0 * scale)),
+                //            null,
+                //            _gus.Name,
+                //            _gus.Sprite.Texture,
+                //            _gus.GetHeadShot(),
+                //            2f);
+
+                //    texture.draw(Game1.spriteBatch);
+
+                //    if (texture.containsPoint(Game1.getMouseX(), Game1.getMouseY()))
+                //    {
+                //        IClickableMenu.drawHoverText(
+                //            Game1.spriteBatch,
+                //            "Gus is selling " + Game1.dishOfTheDay.DisplayName + " recipe today!",
+                //            Game1.dialogueFont);
+                //    }
+                //}
             }
         }
 
         private void OnRenderedHud(object sender, RenderedHudEventArgs e)
         {
-            // Show text on hover
             if (_drawQueenOfSauceIcon.Value && !Game1.IsFakedBlackScreen() && (_icon.Value?.containsPoint(Game1.getMouseX(), Game1.getMouseY()) ?? false))
             {
                 IClickableMenu.drawHoverText(Game1.spriteBatch, _helper.SafeGetString(LanguageKeys.TodaysRecipe) + _todaysRecipe, Game1.dialogueFont);
             }
         }
+        #endregion
 
-        public void Dispose()
-        {
-            ToggleOption(false);
-        }
-
-        /// <summary>Raised after the game begins a new day (including when the player loads a save).</summary>
-        /// <param name="sender">The event sender.</param>
-        /// <param name="e">The event arguments.</param>
-        private void OnDayStarted(object sender, DayStartedEventArgs e)
-        {
-            CheckForNewRecipe();
-        }
-
-        private void OnSaveLoaded(object sender, SaveLoadedEventArgs e)
+        #region Logic
+        private void LoadRecipes()
         {
-            CheckForNewRecipe();
+            if (_recipes.Count == 0)
+            {
+                _recipes = Game1.content.Load<Dictionary<string, string>>("Data\\TV\\CookingChannel");
+                foreach (var next in _recipes)
+                {
+                    string[] values = next.Value.Split('/');
+                    if (values.Length > 1)
```

Comment:
You should also update this condition because the highest index used is 2, not 1 anymore!
```suggestion
                    if (values.Length > 2)
```